src/lava/lib/optimization/solvers/generic/processes.py

```diff
@@ -0,0 +1,290 @@
+            quadratic_component = coeff * np.logical_not(np.eye(*coeff.shape))
+            if 1 in vars.keys():
+                vars[1].init = vars[1].init + linear_component
+            else:
+                vars[1] = Var(
+                    shape=linear_component.shape, init=linear_component
+                )
+            init = -quadratic_component
+        vars[rank] = Var(shape=coeff.shape, init=init)
+    return vars
+
+
+def _in_ports_from_coefficients(
+    coefficients: CoefficientTensorsMixin,
+) -> ty.List[AbstractProcessMember]:
+    in_ports = [
+        InPort(shape=coeff.shape) for coeff in coefficients.coefficients
+    ]
+    return in_ports
+
+
+class SolverProcessBuilder:
+    def __init__(self):
+        self._constructor = None
+
+    def create_constructor(self, problem):
+        """Create constructor for dynamically created
+        OptimizationSolverProcess class."""
+
+        def constructor(
+            self,
+            name: ty.Optional[str] = None,
+            log_config: ty.Optional[LogConfig] = None,
+        ) -> None:
+            super(type(self), self).__init__(name=name, log_config=log_config)
+            self.problem = problem
+            if not hasattr(problem, "variables"):
+                raise Exception(
+                    "An optimization problem must contain " "variables."
+                )
+            if hasattr(problem.variables, "continuous") or isinstance(
+                problem.variables, ContinuousVariables
+            ):
+                self.continuous_variables = Var(
+                    shape=(problem.variables.continuous.num_vars, 2)
+                )
+            if hasattr(problem.variables, "discrete") or isinstance(
+                problem.variables, DiscreteVariables
+            ):
+                self.discrete_variables = Var(
+                    shape=(
+                        problem.variables.num_variables,
+                        # problem.variables.domain_sizes[0]
+                    )
+                )
+            self.cost_diagonal = None
+            if hasattr(problem, "cost"):
+                self.cost_coefficients = _vars_from_coefficients(
+                    problem.cost.coefficients
+                )
+                self.cost_diagonal = -problem.cost.coefficients[
+                    2].diagonal()
+            self.variable_assignment = Var(
+                shape=(problem.variables.num_variables,)
+            )
+            self.optimality = Var(shape=(1,))
+            self.feasibility = Var(shape=(1,))
+
+        self._constructor = constructor
+
+    @property
+    def solver_process(self) -> AbstractProcess:
+        """Hierarchical process implementing an optimization solver."""
+        SolverProcess = type(
+            "OptimizationSolverProcess",
+            (AbstractProcess,),
+            {"__init__": self._constructor},
+        )
+        return SolverProcess()
+
+
+class ContinuousVariablesProcess(AbstractProcess):

```

Comment:
Are those classes the same as the ones in src/lava/lib/optimization/solvers/generic/hierarchical_processes.py ?